### ISSUE TITLE
TopIcons-plus does not display icon XXX

### ISSUE THREAD

**Issue by phocean** [phocean/TopIcons-plus#47]
I received several of such similar bug reports, so here is a catch all issue to make things clear.

Gnome has dropped status icon since GTK 3.14. It is a design decision, thus the ugly legacy tray on the bottom left of the screen.

Here are a few references to support this statement:

https://developer.gnome.org/gtk3/stable/GtkStatusIcon.html
https://bugzilla.gnome.org/show_bug.cgi?id=749074
https://bugzilla.gnome.org/show_bug.cgi?id=758677
https://bugzilla.redhat.com/show_bug.cgi?id=1389875

So, now, as applications get slowly updated or compiled with newer version of GTK, breakages happens more and more.

So how are applications supposed to work now ? Basically, they are supposed to use the notification system to show status and controls, which imply they have a daemon running in background and the GUI communicates over DBUS.
Such a model is nice and works well, see how the Lollypop music player is doing, for instance.
The problem is with legacy apps, however...

In all that, TopIcons-plus just make a cosmetic change, moving the Gnome tray to somewhere else.
It simply relies on the exact same GTK API.

So do not expect that TopIcons-plus magically fixes the GTK issue or acts as a replacement.
A replacement would require much heavier modifications, so you would rather make voice and comment aforementioned Gnome issues, or look for an alternative DE if you are not satisfied (see Solus, for instance).

So, to summarize, here is what you can do:

- do not report a bug here.

- comment/vote on existing Gnome issues. Let them know your feeling.

- create bug reports to applications that do not integrate well with Gnome, so that they can support the new design.

- use another DE.

And yes, at some point TopIcons will be useless. It is only useful to not being polluted with a stupid tray when using old applications.
It has no usefulness for actively maintained applications.

**Comment by phocean** [comment]
In some cases, this is helpful:
`GDK_BACKEND=x11 telegram`

**Comment by audreytoskin** [comment]
Certainly as of GNOME 3.22, there are enough applications on my system that haven't yet been updated to use Wayland, that I find it easier to just switch my login session to GNOME *with X*. When first logging into GNOME, before you enter your password, you'd click the gear icon to change the session type.

This won't be necessary forever, but it's easier to do it this way for now.
